### PR TITLE
cov: add tests for ignite.contrib.metrics

### PR DESCRIPTION
**Pull request by ydcjeff**
Part of #1790 

Description: Add tests for `ignite.contrib.metrics`

Put shape checking to the end

Check list:

- [x] New tests are added (if a new feature is added)
- [ ] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

File: tests/ignite/contrib/metrics/regression/test_mean_absolute_relative_error.py [hunk 1]
```diff
@@ -10,17 +11,11 @@
 def test_wrong_input_shapes():
     m = MeanAbsoluteRelativeError()
 
-    with raises(ValueError, match=r"Input data shapes should be the same, but given"):
-        m.update((torch.rand(4, 1, 2), torch.rand(4, 1)))
+    with pytest.raises(ValueError, match=r"Input data shapes should be the same, but given"):
```
Comment (ydcjeff) [inline tests/ignite/contrib/metrics/regression/test_mean_absolute_relative_error.py:14]:
```suggestion
    with raises(ValueError, match=r"Input data shapes should be the same, but given"):
```

File: tests/ignite/contrib/metrics/regression/test_mean_absolute_relative_error.py [hunk 2]
```diff
@@ -10,17 +11,11 @@
...
 
-    with raises(ValueError, match=r"Input data shapes should be the same, but given"):
-        m.update((torch.rand(4, 1), torch.rand(4, 1, 2)))
-
-    with raises(ValueError, match=r"Input data shapes should be the same, but given"):
-        m.update((torch.rand(4, 1, 2), torch.rand(4,),))
-
-    with raises(ValueError, match=r"Input data shapes should be the same, but given"):
-        m.update((torch.rand(4,), torch.rand(4, 1, 2),))
+    with pytest.raises(ValueError, match=r"Input data shapes should be the same, but given"):
```
Comment (ydcjeff) [inline tests/ignite/contrib/metrics/regression/test_mean_absolute_relative_error.py:17]:
```suggestion
    with raises(ValueError, match=r"Input data shapes should be the same, but given"):
```

File: tests/ignite/contrib/metrics/regression/test_mean_absolute_relative_error.py [hunk 3]
```diff
@@ -1,4 +1,5 @@
 import numpy as np
+import pytest
```
Comment (ydcjeff) [inline tests/ignite/contrib/metrics/regression/test_mean_absolute_relative_error.py:2]:
```suggestion
```

**Review by sdesrozis — APPROVED** [review]
@ydcjeff Thank you ! LGTM

File: tests/ignite/contrib/metrics/test_roc_curve.py [hunk 4]
```diff
@@ -8,6 +11,20 @@
...
+    with patch.dict("sys.modules", {"sklearn.metrics": None}):
+        yield sklearn
+
+
+def test_no_sklearn(mock_no_sklearn):
+    with pytest.raises(RuntimeError, match=r"This contrib module requires sklearn to be installed"):
+        y = torch.tensor([1, 1])
+        roc_curve = RocCurve()
+        roc_curve.update((y, y))
+        roc_curve.compute()
```
Comment (ydcjeff) [inline tests/ignite/contrib/metrics/test_roc_curve.py:25]:
```suggestion
        RocCurve()
```

**Pull request merged**